### PR TITLE
phpstan fix notification class issues

### PR DESCRIPTION
**Pull request by dameert**

### CODE REVIEW

File: Service/NotificationService.php [hunk 1]
```diff
@@ -154,15 +157,9 @@ public function setStatus(Notification $notification, $status, $level = 'notice'
     
     /**
      * Call addNotification when click on a request
-     *
-     * @param unknown $template
-     * @param unknown $revision
      */
-    public function addNotification($templateId, $revision, $environment)
+    public function addNotification(string $templateId, Revision $revision, Environment $environment)
```
Comment (dameert) [inline Service/NotificationService.php]: code calling this function is compatible with strong typing (string, Revision & Environment)

Comment (theus77) [reply Service/NotificationService.php]: I think that a template id id an integer, not a string. Or it's a string but converted to a integer somewhere

File: Twig/AppExtension.php [hunk 2]
```diff
@@ -1060,6 +1060,11 @@ public function searchesList($username)
         return $searches;
     }
 
+    /**
+     * PHPStan complains about this, we should really consider using the {% dump %} functionality and Symfony StackTrace.
+     * And dump got added in twig 1.5:
+     * https://twig.symfony.com/doc/1.x/functions/dump.html
+     */
     public function dump($object)
```
Comment (theus77) [inline Twig/AppExtension.php]: Agree, it can be flagged as deprecated

File: Service/NotificationService.php (hunk 1 above)
Comment (dameert) [reply Service/NotificationService.php]: @theus77 
Looks like there is a problem on line 170:
$template = $repository->findOneById($templateId);

The function findOneById does not exist in the repositories.

Comment (dameert) [reply Service/NotificationService.php]: Nvm i found the magic in __call of EntityRepository (doctrine)

File: Twig/AppExtension.php (hunk 2 above)
Comment (dameert) [reply Twig/AppExtension.php]: I will do this in another pull request

**Pull request merged**